File: practical-bdd/java/pages/ContactPage.java
package pages;

import org.junit.jupiter.api.Assertions;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.Select;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;

public class ContactPage extends BasePage {

    @FindBy(css = "input[name='your-stay-location'][value='Ja']")
    private WebElement stayLocationJa;

    @FindBy(css = "input[name='your-stay-location'][value='Nee']")
    private WebElement stayLocationNee;

    @FindBy(css = "input[name='your-name']")
    private WebElement name;

    @FindBy(css = "input[name='your-email']")
    private WebElement email;

    @FindBy(css = "input[name='your-subject']")
    private WebElement subject;

    @FindBy(css = "textarea[name='your-message']")
    private WebElement message;

    @FindBy(css = "select#your-department")
    private WebElement department;

    @FindBy(css = "input#send-button")
    private WebElement sendButton;

    private final String messageSentIndicatorLocator = "form.sent div.wpcf7-response-output";

    @FindBy(css = messageSentIndicatorLocator)
    private WebElement messageSentIndicator;

    public ContactPage(WebDriver driver) {
        super(driver);
    }

    /**
     * Sends a contact message by filling out the form and clicking the send button.
     *
     * @param stayAtPark the stay location value, either "ja" or "nee"
     * @param name the name of the sender
     * @param email the email address of the sender
     * @param subject the subject of the message
     * @param message the content of the message
     * @param department the department to which the message is sent
     */
    public void sendContactMessage(
            String stayAtPark,
            String name,
            String email,
            String subject,
            String message,
            String department
    ) {
        selectStayAtPark(stayAtPark);
        this.name.sendKeys(name);
        this.email.sendKeys(email);
        this.subject.sendKeys(subject);
        this.message.sendKeys(message);
        selectDepartment(department);
        sendButton.click();
    }

    /**
     * Checks if the contact message was sent successfully by verifying the visibility of the message sent indicator.
     *
     * @return true if the message sent indicator is displayed, false otherwise
     */
    public boolean isMessageSent() {
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(30));
        wait.until(ExpectedConditions.visibilityOfElementLocated(By.cssSelector(messageSentIndicatorLocator)));
        return messageSentIndicator.isDisplayed();
    }

    private void selectStayAtPark(String stayAtPark) {

        switch (stayAtPark.toLowerCase()) {
            case "ja" -> stayLocationJa.click();
            case "nee" -> stayLocationNee.click();
            default -> Assertions.fail("invalid stay at park value " + stayAtPark);
        }
    }

    private void selectDepartment(String department) {
        Select selectDepartment = new Select(this.department);
        selectDepartment.selectByValue(department);
    }

}

File: practical-bdd/typescript/pages/contact.page.ts
import { Page } from '@playwright/test';
import { BasePage } from './base.page';

export class ContactPage extends BasePage {
  private readonly stayLocationJa = this.page.locator('input[name="your-stay-location"][value="Ja"]');
  private readonly stayLocationNee = this.page.locator('input[name="your-stay-location"][value="Nee"]');
  private readonly nameField = this.page.locator('input[name="your-name"]');
  private readonly emailField = this.page.locator('input[name="your-email"]');
  private readonly subjectField = this.page.locator('input[name="your-subject"]');
  private readonly messageField = this.page.locator('textarea[name="your-message"]');
  private readonly departmentSelect = this.page.locator('select#your-department');
  private readonly sendButton = this.page.locator('input#send-button');
  private readonly messageSentIndicator = this.page.locator('div.wpcf7-response-output');

  constructor(page: Page) {
    super(page);
  }

  async sendContactMessage(
    stayAtPark: string,
    name: string,
    email: string,
    subject: string,
    message: string,
    department: string,
  ): Promise<void> {
    await this.selectStayAtPark(stayAtPark);
    await this.nameField.fill(name);
    await this.emailField.fill(email);
    await this.subjectField.fill(subject);
    await this.messageField.fill(message);
    await this.departmentSelect.selectOption({ label: department });
    await this.sendButton.click();
  }

  async isMessageSent(): Promise<boolean> {
    await this.messageSentIndicator.waitFor({ state: 'visible', timeout: 30_000 });
    return this.messageSentIndicator.isVisible();
  }

  private async selectStayAtPark(stayAtPark: string): Promise<void> {
    switch (stayAtPark.toLowerCase()) {
      case 'ja':
        await this.stayLocationJa.click();
        break;
      case 'nee':
        await this.stayLocationNee.click();
        break;
      default:
        throw new Error(`Invalid stay at park value: ${stayAtPark}`);
    }
  }
}
